src/azure-cli/azure/cli/command_modules/cdn/commands.py

```diff
@@ -207,3 +275,100 @@ def load_command_table(self, _):
+
+    with self.command_group('afd endpoint', cdn_afd_endpoints_sdk, is_preview=True) as g:
+        g.show_command('show', 'get')
+        g.command('list', 'list_by_profile')
+        g.command('purge', 'purge_content', supports_no_wait=True)
+        g.command('delete', 'delete', confirmation=True)
+
+        g.custom_command('update', 'update_afd_endpoint', client_factory=cf_afd_endpoints)
+        g.custom_command('create', 'create_afd_endpoint', client_factory=cf_cdn,
+                         doc_string_source='azure.mgmt.cdn.models#AFDEndpoint')
+
+    with self.command_group('afd origin-group', cdn_afd_origin_group_sdk, is_preview=True) as g:
+        g.show_command('show', 'get')
+        g.command('list', 'list_by_profile')
+        g.custom_command('create', 'create_afd_origin_group', client_factory=cf_afd_origin_groups)
+        g.custom_command('update', 'update_afd_origin_group', client_factory=cf_afd_origin_groups)
+        g.command('delete', 'delete', confirmation=True)
+
+    with self.command_group('afd origin', cdn_afd_origin_sdk, is_preview=True) as g:
+        g.show_command('show', 'get')
+        g.command('list', 'list_by_origin_group')
+        g.custom_command('create', 'create_afd_origin', client_factory=cf_afd_origins)
+        g.custom_command('update', 'update_afd_origin', client_factory=cf_afd_origins)
+        g.command('delete', 'delete', confirmation=True)
+
+    with self.command_group('afd route', cdn_afd_route_sdk, is_preview=True) as g:
+        g.show_command('show', 'get')
+        g.command('list', 'list_by_endpoint')
+        g.custom_command('create', 'create_afd_route', client_factory=cf_afd_routes)
+        g.custom_command('update', 'update_afd_route', client_factory=cf_afd_routes)
+        g.command('delete', 'delete', confirmation=True)
+
+    with self.command_group('afd rule-set', cdn_afd_rule_set_sdk, is_preview=True) as g:
+        g.show_command('show', 'get')
+        g.command('list', 'list_by_profile')
+        g.custom_command('create', 'create_afd_rule_set', client_factory=cf_afd_rule_sets)
+        g.command('delete', 'delete', confirmation=True)
+
+    with self.command_group('afd rule', cdn_afd_rule_sdk, is_preview=True) as g:
+        g.show_command('show', 'get')
+        g.command('list', 'list_by_rule_set')
+        g.custom_command('create', 'create_afd_rule', client_factory=cf_afd_rules)
+        g.command('delete', 'delete', confirmation=True)
+
+    with self.command_group('afd rule condition', cdn_afd_rule_sdk, is_preview=True) as g:
+        g.custom_command('add', 'add_afd_rule_condition', client_factory=cf_afd_rules,
+                         doc_string_source='azure.mgmt.cdn.models#Rule')
+        g.custom_command('remove', 'remove_afd_rule_condition', client_factory=cf_afd_rules,
+                         doc_string_source='azure.mgmt.cdn.models#Rule')
+        g.custom_command('list', 'list_afd_rule_condition', client_factory=cf_afd_rules)
```

Comment:
Why not use custom_command_type=XXX property in command_group? If you define a custom command type with client_factory = cf_afd_rules, you don't need to write it again in g.custom_command function.